File: Leetcode/1640. Check Array Formation Through Concatenation/solution.go
func canFormArray(arr []int, pieces [][]int) bool {
    index := make(map[int][]int)
    
    for _, list := range pieces {
        index[list[0]] = list
    }
    
    result := make([]int, 0)
    
    for _, num := range arr {
        if list, ok := index[num]; ok {
            result = append(result, list...)
        }
    }
    
    return reflect.DeepEqual(result, arr)
}

File: Leetcode/1640. Check Array Formation Through Concatenation/solution.py
class Solution:
    def canFormArray(self, arr: List[int], pieces: List[List[int]]) -> bool:
        index = {}

        for piece in pieces:
            index[piece[0]] = piece

        concated = []
        for elem in arr:
            if index.get(elem):
                concated.extend(index.get(elem))

        return concated == arr
